packages/api/src/utils/client/httpClient.ts

```diff
@@ -133,9 +132,28 @@ export class HttpClient implements IHttpClient {
       if (!urlOpts.baseUrl) {
         throw Error(`HttpClient.urls[${i}] is empty or undefined: ${urlOpts.baseUrl}`);
       }
-      if (!isValidHttpUrl(urlOpts.baseUrl)) {
+      const url = validHttpUrl(urlOpts.baseUrl);
```

Comment:
If we don't strictly need to extract the credentials before validating the URL I think adding this additional complexity to authorization header managed it not ideal.

I also noticed the refactor we are doing in https://github.com/ChainSafe/lodestar/pull/6080 allows to change the URL per request which means this implementation won't work.

We have extracted the set the authorization header there already into a separate function
https://github.com/ChainSafe/lodestar/blob/13b65fc3c0556e25a6325d4e6f3b69fdf748a54e/packages/api/src/utils/headers.ts#L87

Will make sure we don't have a regression in the refactor, I definitely should look there how we can maybe avoid base64 encoding header on each request but even then, I haven't benchmarked it but I think it's probably negligible.

Reply:
Sounds like it's preferable to simply call `decodeURIComponent` on username/password before base64 encoding until #6080 is released, in that case. I'll drop the first commit.